articles/quickstart/spa/angular-beta/_includes/_centralized_login.md

```diff
@@ -0,0 +1,459 @@
+
+Make sure that the domain and client ID values are correct for the application that you want to connect with.
+
+## Create a Navigation Bar Component
+
+If you do not already have a logical place to house login and logout buttons within your application, create a new component to represent a navigation bar that can also hold some UI to log in and log out:
+
+```bash
+ng generate component navbar
+```
+
+Open the `src/app/navbar/navbar.component.ts` file and replace its contents with the following:
+
+```js
+import { Component, OnInit } from '@angular/core';
+import Auth0Client from '@auth0/auth0-spa-js/dist/typings/Auth0Client';
+import { AuthService } from 'src/app/auth.service';
+
+@Component({
+  selector: 'app-navbar',
+  templateUrl: './navbar.component.html',
+  styleUrls: ['./navbar.component.css']
+})
+export class NavbarComponent implements OnInit {
+  isAuthenticated = false;
+  profile: any;
+
+  private auth0Client: Auth0Client;
+
+  /**
+   * Constructor - inject the AuthService class
+   */
+  constructor(private authService: AuthService) {}
+
+  /**
+   * Handle component initialization
+   */
+  async ngOnInit() {
+    // Get an instance of the Auth0 client
+    this.auth0Client = await this.authService.getAuth0Client();
+
+    // Watch for changes to the isAuthenticated state
+    this.authService.isAuthenticated.subscribe(value => {
+      this.isAuthenticated = value;
+    });
+
+    // Watch for changes to the profile data
+    this.authService.profile.subscribe(profile => {
+      this.profile = profile;
+    });
+  }
+
+  /**
+   * Logs in the user by redirecting to Auth0 for authentication
+   */
+  async login() {
+    await this.auth0Client.loginWithRedirect({
+      redirect_uri: `<%= "${window.location.origin}" %>/callback`
+    });
+  }
+
+  /**
+   * Logs the user out of the applicaion, as well as on Auth0
+   */
+  logout() {
+    this.auth0Client.logout({
+      client_id: this.authService.config.client_id,
+      returnTo: window.location.origin
+    });
+  }
+}
+```
+
+Notice that the `AuthService` class you created in the previous section is being injected into the component through the constructor. This allows you to get an instance of the client, as well as react to changes in authentication state.
+
+The main setup work is being carried out inside `ngOnInit`, where the RxJS `BehaviorSubject` instances are being used. Whenever the values of these change, the Navbar component will react to those changes and provide an updated UI.
+
+Functions are provided to log in and log out the user using `loginWithRedirect` and `logout` directly on the Auth0 client. In the login scenario, a URI is provided that indicates where Auth0 should redirect to once authentication is complete. This URL must be registered with Auth0 as an Allowed Callback URL in order for login to work.
```

Comment:
> This URL must be registered with Auth0 as an Allowed Callback URL in order for login to work.

Maybe some guidance on exactly where this is located in Auth0? Can we link to the Application settings here?

Reply:
Have added a link in here straight to the settings page.